upgrade_tests/drop_compact_storage_upgrade_test.py

```diff
@@ -0,0 +1,114 @@
+        for node in [node1, node2]:
+            node.drain()
+            node.watch_log_for("DRAINED")
+            node.stop(wait_other_notice=False)
+
+            self.set_node_to_current_version(node)
+            node.start(wait_for_binary_proto=True)
+
+        session = self.patient_cql_connection(node1)
+        try:
+            session.execute("ALTER TABLE drop_compact_storage_test.test DROP COMPACT STORAGE")
+            self.fail("No exception has been thrown")
+        except InvalidRequest as e:
+            assert "Cannot DROP COMPACT STORAGE until all SSTables are upgraded, please run `nodetool upgradesstables` first." in str(e)
+
+        for node in [node1, node2]:
+            node.nodetool("upgradesstables")
+
+        session.execute("ALTER TABLE drop_compact_storage_test.test DROP COMPACT STORAGE")
+
+    def test_drop_compact_storage_failure(self):
+        cluster = self.cluster
+        cluster.populate(2)
+        node1, node2 = cluster.nodelist()
+        cluster.set_install_dir(version="2.1.14")
+        cluster.start(wait_for_binary_proto=True)
+
+        session = self.patient_cql_connection(node1)
+        session.execute(
+            "CREATE KEYSPACE drop_compact_storage_test WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '2'};")
+        session.execute(
+            "CREATE TABLE drop_compact_storage_test.test (a ascii PRIMARY KEY, b ascii, c ascii) WITH COMPACT STORAGE;")
+        for i in range(1, 100):
+            session.execute(
+                "INSERT INTO drop_compact_storage_test.test (a, b, c) VALUES ('{}', '{}', '{}');".format(i, i + 1,
+                                                                                                         i + 2))
+
+        logging.debug("Upgrading to current version")
+        for node in [node1, node2]:
+            node.drain()
+            node.watch_log_for("DRAINED")
+            node.stop(wait_other_notice=False)
+
+            self.set_node_to_current_version(node)
+            node.start(wait_for_binary_proto=True)
+
+        session = self.patient_cql_connection(node1)
+        try:
+            session.execute("ALTER TABLE drop_compact_storage_test.test DROP COMPACT STORAGE")
+            self.fail("No exception has been thrown")
+        except InvalidRequest as e:
+            assert "Cannot DROP COMPACT STORAGE until all SSTables are upgraded, please run `nodetool upgradesstables` first." in str(
+                e)
+
+        node1.nodetool("upgradesstables")
+        thrown = False
+        exception = None
+        try:
+            session.execute("ALTER TABLE drop_compact_storage_test.test DROP COMPACT STORAGE")
+        except Exception as e:
+            exception = e
+            thrown = True
+
+        assert thrown, "No exception has been thrown"
+        assert "Cannot DROP COMPACT STORAGE until all SSTables are upgraded, please run `nodetool upgradesstables` first." in str(
+            exception)
```

Comment:
This could be simplified in the same way we do right before:
```suggestion
        try:
            session.execute("ALTER TABLE drop_compact_storage_test.test DROP COMPACT STORAGE")
            self.fail("No exception has been thrown")
        except InvalidRequest as e:
            assert "Cannot DROP COMPACT STORAGE until all SSTables are upgraded, please run `nodetool upgradesstables` first." in str(e)
```
We could also consider putting this into a separate method, since we repeat it three times.